File: src/test/java/com/datarepublic/testexercise/ebayframework/ItemDetails.java
package com.datarepublic.testexercise.ebayframework;

import java.net.*;

public class ItemDetails
{
    private final String title;
    private final String uri;
    private final String price;
    
    public String title() { return title; };
    public String uri() { return uri; };
    public String price() { return price; };
    
    public ItemDetails(
            String title,
            String uri,
            String price)
    {
        this.title = title;
        this.uri = uri;
        this.price = price;
    }
    
    public String itemId() throws URISyntaxException
    {
        URI itemURI = new URI(uri);
        String itemId = itemURI.getPath().toString();
        itemId = itemId.substring(itemId.lastIndexOf("/")+1);
        return itemId;
    }

}

File: net/TestExerciseEbay/EbayPOMFramework/ItemDetails.cs
using System;

namespace TestExercise.EbayPOMFramework
{
    public class ItemDetails
    {

        public string Title { get; }
        public string Uri { get; }
        public string Price { get; }

        public ItemDetails(
            String title,
            String uri,
            String price)
        {
            this.Title = title;
            this.Uri = uri;
            this.Price = price;
        }

        public string ItemId
        {
            get
            {
                Uri itemURI = new Uri(Uri);
                string itemId = itemURI.AbsolutePath;
                itemId = itemId.Substring(itemId.LastIndexOf("/")+1);
                return itemId;
            }
        }
    }
}